lib/rucio/cli/upload.py

```diff
@@ -0,0 +1,60 @@
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import click
+
+from rucio.cli.bin_legacy.rucio import upload
+from rucio.cli.utils import Arguments
+from rucio.common.config import config_get_float
+
+
+@click.command("upload")
+@click.argument("file-paths", nargs=-1)
+@click.option("--rse", "--rse-name", help="Rucio Storage Element (RSE) name", required=True)
+@click.option("--lifetime", type=int, help="Lifetime of the rule in seconds")
+@click.option("--expiration-date", help="The date when the rule expires in UTC, format: <year>-<month>-<day>-<hour>:<minute>:<second>. E.g. 2022-10-20-20:00:00")
+@click.option("--scope", help="Scope name.")
+@click.option("--impl", type=click.Choice([]), help="Transfer protocol implementation to use (e.g: xrootd, gfal.NoRename, webdav, ssh.Rsync, rclone)")
+# The --no-register option is hidden. This is pilot ONLY. Users should not use this. Will lead to unregistered data on storage!
+@click.option("--no-register", is_flag=True, default=False, hidden=True)
```

Comment:
I think you need to add an additional `"register"` argument here for the tests to pass.